larky/src/test/resources/vendor_tests/iso8583/test_cacao_payload.star

```diff
@@ -0,0 +1,968 @@
+load("@stdlib//binascii", unhexlify="unhexlify", hexlify="hexlify")
+load("@stdlib//builtins", "builtins")
+load("@stdlib//unittest", "unittest")
+load("@vendor//asserts", "asserts")
+load("@vendor//iso8583/decoder", Decoder="decoder")
+load("@vendor//iso8583/encoder", Encoder="encoder")
+
+
+def CacaoTestCase_test_decode_encode():
+    payload = bytes(cacao_payload, encoding='ascii')
+    decoded, encoded = Decoder.decode(payload, cacao_specs)
+    print('decoded:' + str(decoded))
+    print('encoded:' + str(encoded))
+
+    asserts.assert_that(decoded['h']).is_equal_to('ISO025000055')
+    asserts.assert_that(decoded['t']).is_equal_to('0200')
+    asserts.assert_that(decoded['p']).is_equal_to('F63EC41128F18018')
+    asserts.assert_that(decoded['1']).is_equal_to('000000001000012C')
+    asserts.assert_that(decoded['2']).is_equal_to('4111111111111111')
+    asserts.assert_that(decoded['14']).is_equal_to('2503')
+    asserts.assert_that(decoded['35']).is_equal_to('4111111111111111=2503')
+    asserts.assert_that(decoded['43']).is_equal_to('GOOGLE *TEMPORARY HOLDg.co/helppay#   GB')
+    asserts.assert_that(decoded['60']).is_equal_to('VISAB017+0000000')
+    asserts.assert_that(decoded['123']).is_equal_to('')
+    asserts.assert_that(decoded['126']).is_equal_to('& 0000600124! Q100002 9 ! Q200002 09! C000026 603  001          70 100 0! C400012 000000003070! 0400020             Y')
+
+    encoded2_raw, encoded2 = Encoder.encode(decoded, cacao_specs)
+    asserts.assert_that(str(encoded2_raw)).is_equal_to(cacao_payload)
```

Comment:
is there a reason why we have to do `str()` here? what's the original type?

Reply:
`type(encoded2_raw) == "bytearray"`

Reply:
but original payload encoding is ascii, not binary, that's why I use strings in this test

Reply:
Can you verify if it is ascii or latin1 encoding?

Reply:
nope, I will need to double check it with the customer

at least, the test payload they sent to us contained only ascii symbols